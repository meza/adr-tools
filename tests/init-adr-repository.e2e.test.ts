import * as childProcess from 'child_process';
import * as fs from 'fs';
import * as os from 'os';
import * as path from 'path';
/* eslint-disable no-sync */
import { afterEach, beforeEach, describe, expect, it } from 'vitest';

describe('Init an ADR Repository', () => {
  const adr = path.resolve(path.dirname(__filename), '../src/index.ts');
  const command = `npx tsx ${adr}`;

  let adrDirectory: string;
  let workDir: string;

  beforeEach(() => {
    // @ts-ignore
    process.env.ADR_DATE = '1992-01-12';
    workDir = path.resolve(fs.realpathSync(fs.mkdtempSync(path.join(os.tmpdir(), 'adr-'))));
    adrDirectory = path.join(workDir, 'doc/adr');
  });

  afterEach(() => {
    childProcess.execSync(`rimraf ${workDir}`);
  });

  it('should use the default directory', () => {
    childProcess.execSync(`${command} init`, { cwd: workDir });
    const expectedFile: string = path.join(adrDirectory, '0001-record-architecture-decisions.md');
    const expectedLockFile: string = path.join(adrDirectory, '.adr-sequence.lock');
    expect(fs.existsSync(expectedFile)).toBeTruthy();
    expect(fs.existsSync(expectedLockFile)).toBeTruthy();

    const fileContents = fs.readFileSync(expectedFile, 'utf8');
    const lockFileContents = fs.readFileSync(expectedLockFile, 'utf8');
    expect(fileContents).toMatchSnapshot();
    expect(lockFileContents).toEqual('1');
  });

  it('should use an alternate directory', () => {
    const directory: string = path.resolve(path.join(workDir, 'tmp', 'alternative-dir'));
    childProcess.execSync(`${command} init ${directory}`, { cwd: workDir });

    const expectedInitFile: string = path.join(directory, '0001-record-architecture-decisions.md');
    const expectedLockFile: string = path.join(directory, '.adr-sequence.lock');
    expect(fs.existsSync(expectedInitFile)).toBeTruthy();
    expect(fs.existsSync(expectedLockFile)).toBeTruthy();

    const fileContents = fs.readFileSync(expectedInitFile, 'utf8');
    const lockFileContents = fs.readFileSync(expectedLockFile, 'utf8');
    expect(fileContents).toMatchSnapshot();
    expect(lockFileContents).toEqual('1');
  });
});
