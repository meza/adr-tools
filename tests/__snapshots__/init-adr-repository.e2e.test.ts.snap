// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Init an ADR Repository > should use an alternate directory 1`] = `
"# 1. Record architecture decisions

Date: 1992-01-12

## Status

Accepted

## Context

We need to record the architectural decisions made on this project.

## Decision

We will use Architecture Decision Records, as [described by Michael Nygard](https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions).

## Consequences

See Michael Nygard's article, linked above.
For a lightweight ADR toolset, see Nat Pryce's [adr-tools](https://github.com/npryce/adr-tools).
> For a node version of the same tooling, see Meza's [adr-tools](https://github.com/meza/adr-tools).
"
`;

exports[`Init an ADR Repository > should use the default directory 1`] = `
"# 1. Record architecture decisions

Date: 1992-01-12

## Status

Accepted

## Context

We need to record the architectural decisions made on this project.

## Decision

We will use Architecture Decision Records, as [described by Michael Nygard](https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions).

## Consequences

See Michael Nygard's article, linked above.
For a lightweight ADR toolset, see Nat Pryce's [adr-tools](https://github.com/npryce/adr-tools).
> For a node version of the same tooling, see Meza's [adr-tools](https://github.com/meza/adr-tools).
"
`;
